src/components/FeatureSetsPanel/FeatureSetsPanel.js

```diff
@@ -57,23 +60,80 @@ const FeatureSetsPanel = ({
       targetKind => targetKind.name === 'externalOffline'
     )
 
-    if (featureStore.newFeatureSet.metadata.name.length === 0) {
-      return setNameValid(false)
+    if (
+      featureStore.newFeatureSet.metadata.name.length === 0 ||
+      !nameValidationPattern.test(featureStore.newFeatureSet.metadata.name)
+    ) {
+      return setValidation(prevState => ({
+        ...prevState,
+        isNameValid: false
+      }))
     }
 
     if (featureStore.newFeatureSet.spec.source.path.length === 0) {
-      return setUrlValid(false)
+      return setValidation(prevState => ({
+        ...prevState,
+        isUrlValid: false
+      }))
+    }
+
+    if (featureStore.newFeatureSet.spec.source.kind === 'parquet') {
+      const {
+        time_field,
+        start_time,
+        end_time
+      } = featureStore.newFeatureSet.spec.source
+
+      if (
```

Comment:
instead of if {} else where you always set some result, you can do this: 
```suggestion
      setValidation(prevState => ({
          ...prevState,
          isTimeFieldValid: time_field.length === 0,
          isStartTimeValid: start_time.length === 0,
          isEndTimeValid: end_time.length === 0
        }))
```